server/events/apply_requirement_handler.go

```diff
@@ -0,0 +1,50 @@
+package events
+
+import (
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/runtime"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/events/yaml/raw"
+	"github.com/runatlantis/atlantis/server/events/yaml/valid"
+)
+
+//go:generate pegomock generate -m --package mocks -o mocks/mock_apply_handler.go ApplyRequirementHandler
+type ApplyRequirementHandler interface {
+	HandleApplyRequirements(repoDir string, ctx models.ProjectCommandContext) (string, string, bool, error)
+}
+
+type DefaultApplyRequirementHandler struct {
+	PullApprovedChecker runtime.PullApprovedChecker
+	WorkingDir          WorkingDir
+}
+
+func (a *DefaultApplyRequirementHandler) HandleApplyRequirements(repoDir string, ctx models.ProjectCommandContext) (applyOut string, failure string, isApplied bool, err error) {
```

Comment:
The return values are confusing to me.  i get that this was copied directly but i think we should at least make some easy changes.

`applyOut` is specific to projectCommandRunner and this method shouldn't care about it. 

`isApplied` is also confusing to me because of the statement: ApplyRequirementsHandler has requirements have not been satisfied if "isApplied".  

I think the only return value we realistically need here is an error message right? If all apply requirements are satisfied, return nil, if not return an error.

Reply:
That's true. We probably don't need `applyOut` and `isApplied`. However, I think we need the failure message and the error since we're using the failure message and the error in `ProjectResult`. I have removed `applyOut` and `isApplied` from the method. Lmk if you think this can be further improved :)